app/ui/browser-modern/views/browser/location/security-badge.jsx

```diff
@@ -29,19 +29,18 @@ class SecurityBadge extends Component {
   render() {
     return (
       <Btn title="Connection"
-        hidden={this.props.hidden}
-        image={this.props.image}
         imgWidth="16px"
         imgHeight="16px"
-        onClick={this.props.onClick} />
+        image={this.props.image}
+        onClick={this.props.onClick}
+        {...this.props} />
```

Comment:
There's no requirement to propagate `style`, since this isn't a widget. If we followed your logic, all browser frontend components that contain buttons or other widgets should pass in all props to their children; this wouldn't make sense. I don't see why we would need to make an exception for this component.

If you mean `className` instead of `style`, then I see what's going on. But I still disagree that we should pass in all props to the button component (no need for pageId in buttons). Just pass in className.